build/github-tf-plan.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'tf-plan'
+on:
+  push:
+    branches-ignore:
+      - 'development'
+      - 'production'
+      - 'non-production'
+
+env:
+  PROJECT_ID: ${{ secrets.PROJECT_ID }}
+  TF_BACKEND: ${{ secrets.TF_BACKEND }}
+
+jobs:
+  run:
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+    steps:
+      - uses: 'actions/checkout@v3'
+
+      - id: 'auth'
+        uses: 'google-github-actions/auth@v1'
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: ${{ secrets.WIF_PROVIDER_NAME }}
+          service_account: ${{ secrets.SERVICE_ACCOUNT_EMAIL }}
+
+      - uses: 'google-github-actions/setup-gcloud@v1'
+        with:
+          install_components: "beta,terraform-tools"
+
+      - uses: 'hashicorp/setup-terraform@v2'
+        with:
+          terraform_version: '1.3'
+
+      - id: setup
+        shell: bash
+        run: |
+          echo "Adding bucket information to backends"
+          for i in `find -name 'backend.tf'`
+          do
+            sed -r -i "s/UPDATE_ME|UPDATE_PROJECTS_BACKEND/${TF_BACKEND}/" $i
+          done
+
+      - id: plan-validate-all
+        run: |
+          echo "Plan validate all"
+          export BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})
+          ${GITHUB_WORKSPACE}/tf-wrapper.sh plan_validate_all "${BRANCH_NAME}" "${GITHUB_WORKSPACE}/policy-library" "${PROJECT_ID}" "FILESYSTEM" "GITHUB"
```

Comment:
Could we also comment back the output in PR? Something like https://github.com/bharathkkb/tf-git-demo/blob/4a71d335e13e3686bc281febea6bae998dd41259/fs-src/.github/workflows/tf.yaml#L47-L98

We can do this in a follow up

Reply:
basic output added